### DISCUSSION TITLE
Puzzled about syntax

### DISCUSSION THREAD

**Discussion by The-Futurist** [OAI/OpenAPI-Specification#4245] [Q&A] [answered]
I'm familiarising myself with OpenAPI (aka Swagger) and have a question that no amount of web searching has shed light on.

This is what I see when I peruse a generated Doc for some "Model"

![image](https://github.com/user-attachments/assets/12b7e957-caa4-4f94-ba1a-e6ab7bee0f2d)

What is that syntax? does it have a name? Is it simply a generic way to display a structure? It is not JSON, YAML, XML etc but why? why this obscure syntax? Is it just a friendly way of displaying stuff or can it also be consumed, does OpenAPI have the ability to consume that syntax?

**Comment by handrews** [comment]
@Korporal that looks like a format specific to the tool you are using- it is not part of the specification.

**Comment by handrews** [comment]
Closing due to lack of further reply for two weeks- feel free to comment or start a new discussion if there was more to talk about here!

**Reply by The-Futurist** [reply]
Hiw @handrews - I was waiting to see if anyone else had anything to say about this. I encountered it here:

https://sisclientweb-101047.campusnexus.cloud/swagger/ui/?doc=https://sisclientweb-101047.campusnexus.cloud/swagger/docs/queries/academics/Books
i.e.

![image](https://github.com/user-attachments/assets/3ca317fd-c16d-4a11-9854-a0463157fc08)

I only have the most meagre familiarity with Swagger, OpenAPI etc. That syntax seemed like something that is surely standardized, part of _some_ specification - but _what_? Why all that stuff wouldn't be simply expressed as JSON makes no sense to me.

**Reply by handrews** [reply, marked as answer]
That looks like a Swagger tool, correct?  You should ask Swagger support about it.  That format is not part of the OpenAPI Specification, so it must be something that they chose.  (I'm one of the more prolific contributors to the recent 3.0.4 and 3.1.1 releases that we just published, so I'm very familiar with what is and is not in the spec).

**Reply by The-Futurist** [reply]
Thanks Henry, it's odd too that an obvious expert like yourself with far more familiarity than me, has never seen this, it's just a weird things to see such an unusual notation and few people able to shed light on it.

I'm going to visit Swagger's Github now and post a link to this thread. 

Thanks.

**Comment by The-Futurist** [comment]
@handrews  - just FYI:

https://github.com/swagger-api/swagger-ui/discussions/10241****
